apps/backend/src/lib/controller/chip/managed/leaderboard/index.ts

```diff
@@ -240,3 +247,261 @@ ManagedChipClient.prototype.GetTopLeaderboardEntries = async function (
     throw new Error("Failed to get top leaderboard entries");
   }
 };
+
+ManagedChipClient.prototype.SubmitProofJob = async function (
+  username: string,
+  jobId: string
+) {
+  await this.prismaClient.proof.create({
+    data: {
+      username,
+      jobId,
+    },
+  });
+};
+
+ManagedChipClient.prototype.PollProofResults = async function () {
+  // Get all incomplete proof jobs
+  const incompleteProofs = await this.prismaClient.proof.findMany({
+    where: {
+      jobCompleted: false,
+    },
+  });
+
+  if (incompleteProofs.length === 0) {
+    return;
+  }
+
+  // Setup API instance similar to tap.tsx
+  const apiConfig: ConfigurationParameters = {
+    basePath: "https://csn-devcon.taceo.io",
+    accessToken: "ASV9PkXpy76KRFtmcQeaLbxT75grdilY",
+  };
+  const configuration = new Configuration(apiConfig);
+  const apiInstance = new JobApi(configuration);
+
+  const completedJobs: {
+    jobId: string;
+    result: JobResult;
+    username: string;
+    createdAt: Date;
+  }[] = [];
+
+  // Check each incomplete proof
+  await Promise.all(
+    incompleteProofs.map(async (proof) => {
+      try {
+        const result = await pollJobResult(apiInstance, proof.jobId);
+
+        if (result) {
+          // Job completed successfully
+          console.log(result);
+          completedJobs.push({
+            result,
+            jobId: proof.jobId,
+            username: proof.username,
+            createdAt: proof.createdAt,
+          });
+        }
+        // If undefined, job is still processing
+      } catch (error) {
+        console.error(`Error polling job ${proof.jobId}:`, error);
+      }
+    })
+  );
+
+  // Update all completed jobs
+  if (completedJobs.length > 0) {
+    const processJobs = completedJobs.map(async (job) => {
+      if (!job.result.publicInput) {
+        return;
+      }
+
+      // parse all properties of job and verify correct public key
+      const publicInput = JSON.parse(job.result.publicInput);
+      const sigNullifier = publicInput[0];
+      const pubkeyNullifier = publicInput[1];
+      const pubkeyNullifierRandomnessHash = publicInput[2];
```

Comment:
NOTE to self to figure out what indices 3 to 7 are...

Reply:
hahahaha